roles/utils_build_push_image/files/build-push-image.sh

```diff
@@ -0,0 +1,60 @@
+#! /bin/bash
+
+function help() {
+    echo "Options:
+        -h: Help
+        -n: Name of image
+        -t: Tag for image
+        -d: Path/Name of Dockerfile
+        -g: Git repo containing Dockerfile
+        -p: Path/Name of git repo Dockerfile
+        -q: Quay.io Org/Repo
+        -a: Authfile for quay.io
+    "
+}
+
+while getopts n:t:g:d:p:q:a:h flag
+do
+    case "${flag}" in
+        h) help
+           exit 0 ;;
+        n) name=${OPTARG};;
+        t) tag=${OPTARG};;
+        g) repo=${OPTARG};;
+        d) dockerfile=${OPTARG};;
+        p) path=${OPTARG};;
+        q) quay="quay.io/${OPTARG}";;
+        a) authfile=${OPTARG};;
+        *) exit 1 ;;
+    esac
+done
+
+if [[ -z $repo && -z $dockerfile ]] ; then
+    echo "Either a git repo (-g) or Dockerfile (-d) is required"
+    exit 1
+elif [[ -n $repo && -n $dockerfile ]] ; then
+    echo "Cannot have both -g and -d"
+    exit 1
+elif [[ -n $repo && -z $path ]] ; then
+    echo -- "Must supply path to Dockerfile within git repo (-p)"
+    exit 1
+elif [[ -z $name || -z $tag ]] ; then
+    echo -- "Image name (-n) and tag (-t) required"
+    exit 1
+fi
+
+if [[ -n $dockerfile ]] ; then
+    podman build -t $name:tag -f $dockerfile
+elif [[ -n $repo ]] ; then
+    git clone --depth 1 $repo repo
```

Comment:
you need to provide a `ref` here, eg `--branch $ref`, so that you can allow cloning another branch/tag/commit than the default `master` branch

and can you add a flag to make it shallow: `--depth 1` I think